lsp-types/src/Language/LSP/Types/Uri.hs

```diff
@@ -190,21 +196,24 @@ instance Hashable NormalizedFilePath where
   hash (NormalizedFilePath uri _) = hash uri
   hashWithSalt salt (NormalizedFilePath uri _) = hashWithSalt salt uri
 
-instance IsString NormalizedFilePath where
-    fromString = toNormalizedFilePath
+filePathToNormalizedFilePath :: MonadThrow m => FilePath -> m NormalizedFilePath
+filePathToNormalizedFilePath fp = OsPath.fromFilePath fp >>= toNormalizedFilePath
+
+normalizedFilePathToFilePath :: MonadThrow m => NormalizedFilePath -> m FilePath
+normalizedFilePathToFilePath nfp = OsPath.toFilePath $ fromNormalizedFilePath nfp
 
-toNormalizedFilePath :: FilePath -> NormalizedFilePath
-toNormalizedFilePath fp = normalizedFilePath nuri nfp
+toNormalizedFilePath :: MonadThrow m => OsPathCompat -> m NormalizedFilePath
```

Comment:
So the failure here ultimately comes from `decodeUtf`, right? Which ultimately represents the fact that a filepath is bytes and a URI should be text. So this is a meaningful failure. We could offer a partial version that really throws, but that also seems like a footgun for us in future.

The only other alternative I can see is to have URI/NormalizedUri also contain an `OsPath` i.e. bytes rather than text. Unclear how bad that would be.